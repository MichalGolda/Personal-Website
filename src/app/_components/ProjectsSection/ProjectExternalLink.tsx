import Image from "next/image";

export type ProjectExternalLinkProps = {
  href: string;
  name: string;
};

const ProjectExternalLink = ({ href, name }: ProjectExternalLinkProps) => (
  <a
    className="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
    href={href}
  >
    {name}
    <Image src="/arrow.webp" alt="arrow" width={32} height={0} />
  </a>
);

export default ProjectExternalLink;
