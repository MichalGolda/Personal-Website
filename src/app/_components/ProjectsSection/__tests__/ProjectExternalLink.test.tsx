import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ProjectExternalLink from "../ProjectExternalLink";

describe("ProjectExternalLink component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <ProjectExternalLink href="https://test.pl" name="test name" />
    );

    const link = container.firstChild;
    const image = link?.lastChild;

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "https://test.pl");
    expect(link?.textContent).toBe("test name");

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("src", "/arrow.svg");
    expect(image).toHaveAttribute("alt", "arrow");
    expect(image).toHaveAttribute("width", "32");
    expect(image).toHaveAttribute("height", "0");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <ProjectExternalLink href="https://test.pl" name="test name" />
    );

    expect(container).toMatchSnapshot();
  });
});
