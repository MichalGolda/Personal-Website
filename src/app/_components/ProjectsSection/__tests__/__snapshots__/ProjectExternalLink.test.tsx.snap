// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectExternalLink component renders unchanged 1`] = `
<div>
  <a
    class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
    href="https://test.pl"
  >
    test name
    <img
      alt="arrow"
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      src="/arrow.svg"
      style="color: transparent;"
      width="32"
    />
  </a>
</div>
`;
